dex/src/main/scala/com/wavesplatform/dex/api/websockets/actors/WsHandlerActor.scala

```diff
@@ -130,22 +125,19 @@ object WsHandlerActor {
                   Behaviors.same
               }
 
-            case command: Command.ProcessClientError =>
-              context.log.debug("Got an error from the client, stopping: {}", command.error)
-              Behaviors.stopped
-
-            case Completed =>
-              context.log.debug("Got a stop request, stopping...")
-              stop(nextPing, pongTimeout)
-
             case command: Command.CloseConnection =>
-              context.log.trace("Got a close request, stopping: {}", command.reason.message.text)
+              context.log.trace("Got CloseConnection: {}", command.reason.message.text)
+              clientRef ! WsError.from(command.reason, time.getTimestamp())
               clientRef ! WsServerMessage.Complete
-              stop(nextPing, pongTimeout)
-          }
-          .receiveSignal {
-            case (_, Terminated(ws)) =>
-              context.log.debug("Client actor stopped, stopping...")
+              cancelSchedules(nextPing, pongTimeout)
+              Behaviors.same // Will receive Completed when WsServerMessage.Complete will be delivered
+
+            case Completed(status) =>
+              status match {
+                case Left(e)  => context.log.debug("Got failure Completed. Stopping...", e)
+                case Right(_) => context.log.debug("Got successful Completed. Stopping...")
+              }
+              cancelSchedules(nextPing, pongTimeout)
```

Comment:
Lines 132 and 140 - only one should survive

Reply:
`Completed` could be even there was no `CloseConnection` command.
But when we receive `CloseConnection` we should not wait and cancel schedules immediately.